### PR TITLE
Add new SwiftUI AnyView modifier extension

### PR DESCRIPTION
**Pull request by jevonmao**
### Motivation

Within the SwiftUI framework, developers frequently need to use an Apple API - the `AnyView` method. It type-erases any SwiftUI `View` and returns a type-erased `AnyView` that does not contain any information about the view hierarchy's structure. 

However, the problem is that the `AnyView` can only be called by using its struct initializer. This syntax requires that the entire view to be erased be passed into the initializer. Many developers find this inconvenient and decrease code readability. For example, notice how the `AnyView` adds a layer to the pyramid of doom:
```swift
struct ContentView: View {
    var body: some View {
        AnyView(
            VStack {
                MapView()
                    .ignoresSafeArea(edges: .top)
                    .frame(height: 300)
                CircleImage()
                    .offset(y: -130)
                    .padding(.bottom, -130)
                VStack(alignment: .leading) {
                    Text("Turtle Rock")
                        .font(.title)
                    HStack {
                        Text("Joshua Tree National Park")
                        Spacer()
                        Text("California")
                    }
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                    Divider()
                    Text("About Turtle Rock")
                        .font(.title2)
                    Text("Descriptive text goes here.")
                }
                .padding()
                Spacer()
            }
        )
    }
}
```
### Solution
The proposed solution is a very simple, short, syntactical sugar that wraps the `AnyView` initializer into a SwiftUI modifier. The modifier syntax is much easier to read and does not disrupt the flow of typing.
Example with SwifterSwift:
```swift
struct ContentView: View {
    var body: some View {
  
        VStack {
            MapView()
                .ignoresSafeArea(edges: .top)
                .frame(height: 300)
            CircleImage()
                .offset(y: -130)
                .padding(.bottom, -130)
            VStack(alignment: .leading) {
                Text("Turtle Rock")
                    .font(.title)
                HStack {
                    Text("Joshua Tree National Park")
                    Spacer()
                    Text("California")
                }
                .font(.subheadline)
                .foregroundColor(.secondary)
                Divider()
                Text("About Turtle Rock")
                    .font(.title2)
                Text("Descriptive text goes here.")
            }
            .padding()
            Spacer()
        }
    }
    .eraseToAnyView()
}
```
or

```
ContentView().eraseToAnyView()
```

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by jevonmao** [comment]
@guykogus and @LucianoPAlmeida please run CI and review my PR. Thank you so much!

File: Sources/SwifterSwift/SwiftUI/ViewExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+public extension View {
+    
+    /**
+     Modifier to erase a SwiftUI `View` type into `AnyView`
+     
+     An `AnyView` allows changing the type of view used in a given view hierarchy. Whenever the type of view used with an AnyView changes, the old hierarchy is destroyed and a new hierarchy is created for the new type.
+     */
+    @inlinable func eraseToAnyView() -> AnyView {
+        AnyView(self)
+    }
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:22]: Add some extra spacing
```suggestion
    }

```

File: Sources/SwifterSwift/SwiftUI/ViewExtensions.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
+//
+//  ViewExtensions.swift
+//  SwifterSwift
+//
+//  Created by Jevon Mao on 6/18/21.
+//  Copyright © 2021 SwifterSwift. All rights reserved.
+//
+
+import Foundation
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:9]: Do you need this import?

File: Sources/SwifterSwift/SwiftUI/ViewExtensions.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import SwiftUI
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:10]: You'll probably need to wrap this in a `#if canImport` statements.

File: Sources/SwifterSwift/SwiftUI/ViewExtensions.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+public extension View {
+    
+    /**
+     Modifier to erase a SwiftUI `View` type into `AnyView`
+     
+     An `AnyView` allows changing the type of view used in a given view hierarchy. Whenever the type of view used with an AnyView changes, the old hierarchy is destroyed and a new hierarchy is created for the new type.
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:18]: This line is too long. Wrap it.

File: Sources/SwifterSwift/SwiftUI/ViewExtensions.swift [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+     */
+    @inlinable func eraseToAnyView() -> AnyView {
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:20]: It's a very verbose name. Maybe something shorter and cleaner like `asAnyView`? Or just `anyView`?

Comment (guykogus) [inline Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:20]: Why is this `@inlinable`? Is that required? Do you know for sure that it's useful/necessary to have here? I think in general we don't use it in the project, but we never decided a standard for if/when to do so.

File: CHANGELOG.md [hunk 6]
```diff
@@ -60,6 +60,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `replacingOccurrences(ofPattern:withTemplate:options:searchRange:)` as a more convenient way to replace patterns. [#901](https://github.com/SwifterSwift/SwifterSwift/pull/901) by [gurgeous](https://github.com/gurgeous)
 - **Measurement**
   - Added `.degrees(_:)`, `arcMinutes(_:)`, `arcSeconds(_:)`, `radians(_:)`, `gradians(_:)` and `revolutions(_:)`  to conveniently initialize measurement with corresponding unit. [#936](https://github.com/SwifterSwift/SwifterSwift/pull/936) by [Shiva Huang](https://github.com/ShivaHuang)
+- **SwiftUI**
+  - Added the  `eraseToAnyView()` modifier as as shortcut for `AnyView()` to conveniently erase any SwiftUI `View` to `AnyView` by [jevonmao](https://github.com/jevonmao)
```
Comment (guykogus) [inline CHANGELOG.md:64]:
```suggestion
  - Added the `eraseToAnyView()` modifier as as shortcut for `AnyView(_:)` to conveniently erase any SwiftUI `View` to `AnyView` by [jevonmao](https://github.com/jevonmao)
```

**Review by guykogus — CHANGES_REQUESTED** [review]
You'll need to update the README and podspec to fully handle this new library section :)

File: Sources/SwifterSwift/SwiftUI/ViewExtensions.swift (hunk 5 above)
Comment (jevonmao) [reply Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:20]: I can't explain the details of `@inlinable` (I think it helps compiler optimizes function call in some cases), I copied this from my own project and the `@inlinable` is just there. Should I remove it for simplicity?

Comment (jevonmao) [reply Sources/SwifterSwift/SwiftUI/ViewExtensions.swift:20]: `anyView` sounds good.

**Comment by jevonmao** [comment]
Let me fix the unit test...

File: Tests/SwiftUITests/ViewExtensionTests.swift [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+@testable import SwifterSwift
+import XCTest
+import SwiftUI
+
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+class ViewExtensionTests: XCTestCase {
+    func testEraseToAnyView() {
+        #if !os(macOS)
+        let typeErasedView = UIHostingController(rootView: TestView().eraseToAnyView())
```
Comment (guykogus) [inline Tests/SwiftUITests/ViewExtensionTests.swift:17]: You need to change this to `anyView()`

File: Tests/SwiftUITests/ViewExtensionTests.swift [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+//  EraseToAnyViewTests.swift
+//  SwifterSwift
+//
+//  Created by Jevon Mao on 6/18/21.
+//  Copyright © 2021 SwifterSwift. All rights reserved.
+//
+
+@testable import SwifterSwift
+import XCTest
+import SwiftUI
```
Comment (guykogus) [inline Tests/SwiftUITests/ViewExtensionTests.swift:11]: You need to wrap this in `#if canImport`, just like the implementation. Look at the other test files to see an example.

**Review by guykogus — CHANGES_REQUESTED** [review]
You need to fix the tests running (commented how).
Also, please add a README entry for the new SwiftUI entry. And maybe update the podspec to include a subspec too. Look in the files and you'll see how other sections were added.

**Comment by jevonmao** [comment]
Yes I will try when I have time, I unfortunately have been busy those past few weeks.

**Pull request closed without merging**